### PR TITLE
Add output class with tests

### PR DESCRIPTION
**Pull request by avanbrunt-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been added that prove the fix is effective or that the feature works.
- [x] New and existing tests pass locally with the changes.
- [x] Code follows the style guidelines of this project (PEP8, clean code).
- [x] Linter has passed locally and any fixes were made for failures.
- [x] A self-review of the code has been done.

## Pull request type

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes (not tied to bugs/features)
- [ ] Other (please describe):

## What is the ticket or issue number?

CBAPI-4629

## Pull Request Description

Add support for currently documented output locations

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How Has This Been Tested?

Unit/Functional tests

### CODE REVIEW

File: src/cbc_syslog/util/output.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+                if "tls" in type:
+                    context = ssl.create_default_context()
+                    context.load_verify_locations(self.output_params["ca_cert"])
+
+                    if "cert" in self.output_params:
+                        context.load_cert_chain(self.output_params["cert"],
+                                                keyfile=self.output_params["key"],
+                                                password=self.output_params["key_password"])
+
+                    if not self.output_params.get("tls_verify", True):
```
Comment (emitreva-cb) [inline src/cbc_syslog/util/output.py:77]: Is the type of the value **string** or **bool**? Based on the example - `"tls_verify": "",`

Comment (avanbrunt-cb) [reply src/cbc_syslog/util/output.py:77]: bool and I'll update the example for context on the types

File: src/cbc_syslog/util/output.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    def __init__(self, output_params):
+        """
+        Initialize the Output object.
+
+        Args:
+            output_params (dict): The output configuration
+
+        Example:
+            {
```
[the remark this answers is not in the document]
Comment (avanbrunt-cb) [reply src/cbc_syslog/util/output.py:35]: The intention is to source the outputs from the config class and not build the configuration manually in python. There is a validation function in the config class which provides detailed log messages. Plus there is a follow up story to write a better readme and getting started document

File: src/cbc_syslog/util/output.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+                "port": str,
+                "tls_verify": bool,
+                "http_headers": dict,
+                "ca_cert": str,
+                "cert": str,
+                "key": str,
+                "key_password": str
+            }
+        """
+        self.output_params = output_params
```
Comment (emitreva-cb) [inline src/cbc_syslog/util/output.py:47]: Should we make a copy of this or it is safe to directly assign it?

Comment (avanbrunt-cb) [reply src/cbc_syslog/util/output.py:47]: It's not intended to be modified in this class if a property needs manipulated then yeah a copy would be made

**Review by kebringer-cb — APPROVED** [review]

**Pull request merged**